doc/languages-frameworks/dotnet.section.md

```diff
@@ -71,8 +71,9 @@ The `dotnetCorePackages.sdk` contains both a runtime and the full sdk of a given
 
 To package Dotnet applications, you can use `buildDotnetModule`. This has similar arguments to `stdenv.mkDerivation`, with the following additions:
 
-* `projectFile` has to be used for specifying the dotnet project file relative to the source root. These usually have `.sln` or `.csproj` file extensions. This can be an array of multiple projects as well.
-* `nugetDeps` takes either a path to a `deps.nix` file, or a derivation. The `deps.nix` file can be generated using the script attached to `passthru.fetch-deps`. This file can also be generated manually using `nuget-to-nix` tool, which is available in nixpkgs. If the argument is a derivation, it will be used directly and assume it has the same output as `mkNugetDeps`.
+* `projectFile` has to be used for specifying the dotnet project file relative to the source root. These usually have `.sln` or `.csproj` file extensions. This can be a list of multiple projects as well.
+* `nugetSha256` is used to specify the hash of the generated nix-compatible lockfile containing all Nuget dependencies. This depends on no floating version ranges being requested upstream, as we cannot deterministically fetch those. If any unstable version range is requested, an error will be thrown and you will need to use `nugetDeps` instead. This option is mutually exclusive with the `nugetDeps` attribute.
+* `nugetDeps` is used to specify a manually generated nix-compatible lockfile, in case we cannot deterministically do so. A script to generate this file is available as `passthru.fetch-deps`. It can also be generated manually using `nuget-to-nix` tool, which is available in nixpkgs. If the argument is a derivation, it will be used directly and assume it has the same output as `mkNugetDeps`. This option is mutually exclusive with the `nugetSha256` attribute.
```

Comment:
```suggestion
* `nugetDeps` is used to specify a manually generated Nix-compatible lockfile, in case we cannot deterministically do so.

  A script to generate this file is available as `passthru.fetch-deps`. It can also be generated manually using `nuget-to-nix` tool, which is available in Nixpkgs. If the argument is a derivation, it will be used directly and assume it has the same output as `mkNugetDeps`. This option is mutually exclusive with the `nugetSha256` attribute.
```